calico_containers/pycalico/netns.py

```diff
@@ -51,6 +52,55 @@ def setup_logging(logfile, level=logging.INFO):
     _log.addHandler(handler)
 
 
+def increment_metrics(namespace):
+    """
+    Increment the metric of any default route that has a metric of 0.
+    :param namespace: The Networking namespace of the container.
+    :return: None. Raises CalledProcessError on error.
+    """
+    def get_metric(route):
+        """
+        Gets the metric of a route.
+        :param route: String representation of a route (full line returned by
+        `ip route`.
+        :return: Integer value representing the metric of the route.
+        """
+        match = re.search('metric\s+(\d)', route)
+        if match:
+            metric = int(match.group(1))
+        else:
+            metric = 0
+        return metric
+
+    with NamedNamespace(namespace) as ns:
+        routes = ns.check_output(["ip", "route"]).split("\n")
+        default_routes = []
+        route_with_metric_0 = False
+        for route in routes:
+            if route.startswith("default"):
+                if "metric" not in route:
```

Comment:
better still, there is an OrderedDict class that might do some of that legwork...